xray/segment.go

```diff
@@ -262,14 +286,18 @@ func (seg *Segment) AddMetadataToNamespace(namespace string, key string, value i
 	return nil
 }
 
-// AddError allows adding an error to the segment.
-func (seg *Segment) AddError(err error) error {
```

Comment:
why refactor this?

Reply:
same of idea of avoiding passing locks around, but also synchronizing properly. we want a public method that safely modifies the segment, but also allow the functionality to be reused in other methods that does its own synchronization